### ISSUE TITLE
aws-s3 provider not working in production

### ISSUE THREAD

**Issue by nirali35** [strapi/strapi#11979]
### Describe the bug

aws-s3 in `plugins.js` is not picked up if placed under `./config/env/production/`

### Steps to reproduce the behavior

1. Place the following file at `./config/env/production/plugins.js`
```
module.exports = ({ env }) => ({
  upload: {
    enabled: true,
    config: {
      provider: "aws-s3",
      providerOptions: {
        accessKeyId: env("AWS_ACCESS_KEY_ID"),
        secretAccessKey: env("AWS_ACCESS_SECRET"),
        region: env("AWS_REGION"),
        params: {
          Bucket: env("AWS_BUCKET"),
        },
      },
    },
  },
});

```

### Expected behavior
Files should be uploaded to AWS S3 in production

### Code snippets
Files are uploaded locally on disk.

### System

- Node.js version:  v14.16.0
- NPM version: 6.14.12
- Strapi version: 4.0.0
- Stack: Heroku

### Additional context

Note other files like `database.js` and `middleware.js` placed under `./config/env/production/` work just fine. Also if the same `plugins.js` file it placed under `./config/` it also works fine. However, we don't want to place it there since we only want to use AWS S3 in production.

**Comment by oaraujocesar** [comment]
This is also happening to me. All the uploaded files keep going to public/uploads directory.

**Comment by derrickmehaffy** [comment]
Fixed in v4.0.2, upgrade to the latest version.

**Comment by wmcmorrow** [comment]
This is also happening to me on version 4.1.2. Uploads are being added to the public/uploads directory.
